### PR TITLE
Improve clarity of tests

### PR DESCRIPTION
**Pull request by kakkoyun**
This PR fixes minor issues that has been introduced by previous PR #30.

cc @brancz

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -254,7 +254,7 @@ func TestControllerConfigmapUpdate(t *testing.T) {
 			shouldBeUpdated: true,
 		},
 		{
-			name:            "ConfigMapWithOtherChange",
+			name:            "ConfigMapWithSameHashringAndOtherChanges",
```
Comment (squat) [inline main_test.go]: what are the `other changes` in this test? Isn't the initial configmap identical, and thus `shouldBeUpdated` is false?

Comment (kakkoyun) [reply main_test.go]: The other changes are the label changes.

>  Isn't the initial configmap identical, and thus `shouldBeUpdated` is false?

Yes, exactly.

Maybe I should be more specific like `ConfigMapWithSameHashringAndLabelChanges`? I can also add test cases like `ConfigMapWithSameHashringAndAnnotationChanges`.

I just want to emphasis and elaborate update behaviour that we have in reconciliation loop. WDYT?

Comment (squat) [reply main_test.go]: Naming is hard. The only issue here is that IMO super long test case names are hard to read and parse for humans especially when they are in camel case. This test is trying to show that the original labels will be kept, right? In that case call it `KeepLabels` and call the other one `DifferentHashring`.

**Comment by squat** [comment]
PTAL @ https://github.com/observatorium/thanos-receive-controller/pull/30#discussion_r332908597

**Comment by kakkoyun** [comment]
@squat Regarding the following comment:

> If you want to trigger a change, then I think the correct order of operations would be to do:
> 1. create klient
> 2. bootstrap initial resources
> 3. start controller
> 4. wait for sync
> 5. compare resources

Since we only want to conduct updates on generated configmap when we have changed hashrings, we are making changes on generated and original configmaps after their initial creation. That's why I've put 2 reconciliation loops in those tests, one after creation and one after updates.

1. create klient 
2. bootstrap initial resources
3. start controller
4. wait for sync
5. **update generated and original resources**
6. wait for sync
7. compare resources

What do you think could we simplify this?

**Comment by squat** [comment]
OK I see what you are saying now but I think that the multiple re-syncs complicate the logic. The tests should be distilled the most minimal set of operations needed to assert our expectations. The point of this test is to ensure that given an existing generated configmap, the resulting configmap will look how we expect, right? So:

1. create klient
2. create initial resources INCLUDING and existing generated configmap (can be more than one func call, does not need to be all in the same func)
3. run controller
4. wait for one single sync, since we want to test that after one sync the resources match our expectations
5. compare

**Pull request merged**